### DISCUSSION TITLE
All White Cables

### DISCUSSION THREAD

**Discussion by malfurion** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#175] [Q&A]
Hello everyone,

I have a Lay Z Spa Miami from 2016 (54123) with 4 Wires and i'm struggling a bit with the Wires to find out wich is 5V, GND and so on.

Does someone have any experience with this model ?
Or is there a good way to find out wich Wire is what for example measuring with a multimeter ?

Here is a Picture of the connector.

![IMG_6077 Large](https://user-images.githubusercontent.com/634939/148903607-7bf9bf37-a3cc-4590-bdae-ae5656c7ac7f.jpeg)

**Comment by chunkysteveo** [comment]
My old Paris 4-wire model has similar cables. You would be best to unscrew the Display unit from the underside of the top which will show you the top of the PCB encased in resin/epoxy. The PCB shows V and G as the outer most wires on my Paris 4-wire. The equates to G being the "short dash" on your picture, and V being the solid white that has small text about the wire on it.

Clock and data are the other two. (X wire and "long dash"). 

The X wire for me goes to pin "3" on the 4-pin header on the custom VisualApproach PCB (next to ground). And the "Long Dash" wire goes to pin "4" which is the bottom most pin on the 4-pin header on the custom PCB. If you don't see correct behaviour between the display and pump unit, those two wires are back to front, just swap them. BUT DONT GET G AND V MESSED UP!

Please open the display and view the PCB from the front to verify they are the same as a Paris.

**Comment by chunkysteveo** [comment]
![20220111_141203](https://user-images.githubusercontent.com/14663002/148962056-e4898fe6-2f2e-4258-9d02-1f753ba7c4c3.jpg)

![20220111_141146](https://user-images.githubusercontent.com/14663002/148962310-fc4e166d-7cb8-4581-8140-c8ad16f2a9b3.jpg)


This may be the best way for you to trace them. This setup works for a 2015/2016 4Wire Paris model.

**Comment by malfurion** [comment]
thx chunkysteveo for the help

i've checked my display and its the same as yours so i've solderd the cables and connected the esp.

My display now works fine with the esp connected and the esp is avalable via the webinterface but i cant control it via the esp and its blinking blue so i think there is some error.

but i cant find any information about the blinking

another strange thing is that the webinterface is showing whats up eg. like pump on but if heater and pump is on the webinterface shows pump and bubbles

and after i choosed take control and setting for example the pump i can hear some clicking inside my pump could be a relay but there is nothing happening

**Comment by chunkysteveo** [comment]
You may need to change the Model definition in the code and re-upload the firmware to the ESP. I had to choose the 54123 to get anything to function. Although if you see my issue on the main github page, i'm working with VA to get my model fully working (54154).

Change the model to 54123 and re-try?

**Reply by tunnus-1** [reply]
I also have a Paris Airjet model (4-wire), but its model number is 54148. While I'm waiting for PCB (v2) and other parts to arrive, I'm trying to find out beforehand the needed settings. Wires look the same as in this thread's first picture, so I'm wondering if I should use 54123 as a model?

**Reply by visualapproach** [reply]
![image](https://github.com/visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA/assets/22987250/105c2f50-88d7-495f-b0d0-bd94a2a3fd31)
you have to try the options that's describe your pump best, and hope it will work. Main diff btwn two seemingly equal pump models is the way they act when turning on stuff. For example; some pumps don't let air and heat be on at the same time (from factory)

**Reply by tunnus-1** [reply]
Got it. Was it probable with my model, that I would only need ESP8266+PCB_V2, but not LLC or any resistors? Also, nowadays all configuration done in UI, and no code commenting / header files editing needed?

Will probably ask more when parts have arrived and first soldering attempts been made...

**Reply by visualapproach** [reply]
> Got it. Was it probable with my model, that I would only need ESP8266+PCB_V2, but not LLC or any resistors? 

It may or may not work. Finding out is the DIY part 😁 

> Also, nowadays all configuration done in UI, and no code commenting / header files editing needed?

That's right!
> 
> 
> Will probably ask more when parts have arrived and first soldering attempts been made...

**Reply by chunkysteveo** [reply]
My Paris 4-wire worked best as a NO54154. However, it no longer works with the new unified firmware, so currently running firmware version 4W_2022-08-03.

**Comment by malfurion** [comment]
@chunkysteveo thanks for the help changing the model helped

the blinking of the esp still exists but i cant find any issues or log entries

although thanks to @visualapproach and all the contributors for the great project

**Comment by chunkysteveo** [comment]
Agree with the lack of info about the blinking LED, but from what I have read on other issues and discussions - the LED blinks every time the Display talks to the CIO - so a blinking LED is a good thing(!) Good job it's hidden as that would do my head in if it was outside the unit!!!! 

haha

**Comment by AmyYummy** [comment]
You need a Multi Funcion [Optical](www.baudcom.cn) fiber Power Meter.
